tests/test_uharfbuzz.py

```diff
@@ -166,3 +166,33 @@ def v_origin_func(font, gid, data):
         hb.shape(blankfont, buf)
         infos = [(pos.y_advance, pos.x_offset, pos.y_offset) for pos in buf.glyph_positions]
         assert infos == expected
+
+    def test_message_func(self, blankfont):
+        string = "edcba"
+        buf = hb.Buffer()
+        buf.add_str(string)
+        buf.guess_segment_properties()
+
+        messages = []
+        infos_trace = []
+        positions_trace = []
+
+        def message(msg):
+            messages.append(msg)
+            infos_trace.append(buf.glyph_infos)
+            positions_trace.append(buf.glyph_positions)
+
+        buf.set_message_func(message)
+        hb.shape(blankfont, buf)
+        gids = [g.codepoint for g in buf.glyph_infos]
+        assert gids == [5, 4, 1, 2, 1]
```

Comment:
it's not clear to what glyph these indexes are referring to (unless one inspects the test font). Can you maybe add some comments?